eth/api.go

```diff
@@ -478,7 +478,9 @@ func (api *PrivateDebugAPI) TraceTransaction(ctx context.Context, txHash common.
 		deadlineCtx, cancel := context.WithTimeout(ctx, timeout)
 		go func() {
 			<-deadlineCtx.Done()
-			tracer.(*ethapi.JavascriptTracer).Stop(&timeoutError{})
+			if deadlineCtx.Err() == context.DeadlineExceeded {
+				tracer.(*ethapi.JavascriptTracer).Stop(&timeoutError{})
+			}
```

Comment:
The problem is that `<-deadlineCtx.Done()` will return when `cancel()` is called, at which point it shouldn't be trying to stop the JS VM. I've changed it to check for `context.Canceled` instead.